modules/programs/timidity.nix

```diff
@@ -0,0 +1,48 @@
+{ config, lib, pkgs, ... }:
+
+let cfg = config.programs.timidity;
+in {
+  meta.maintainers = [ lib.maintainers.amesgen ];
+
+  options.programs.timidity = {
+    enable = lib.mkEnableOption "timidity, a software MIDI renderer";
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.timidity;
+      defaultText = lib.literalExpression "pkgs.timidity";
+      description = "The timidity package to use.";
+    };
+
+    wrappedPackage = lib.mkOption {
+      readOnly = true;
+      type = lib.types.package;
+      description = "Resulting package.";
+    };
+
+    extraConfig = lib.mkOption {
+      type = lib.types.lines;
+      default = "";
+      example = ''
+        soundfont ''${pkgs.soundfont-fluid}/share/soundfonts/FluidR3_GM2-2.sf2
+      '';
+      description = "Extra configuration.";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    home.packages = [ cfg.wrappedPackage ];
+
+    programs.timidity = {
```

Comment:
Does `timidity` work on MacOS?

If not, an assert check needs to be added:

``` nix
{
  assertions = [
    (lib.hm.assertions.assertPlatform "programs.timidity" pkgs
        platforms.linux)
  ];
}
```

Reply:
I have no access to a MacOS machine, but people seem to install it via Homebrew on MacOS: https://formulae.brew.sh/formula/timidity

Reply:
The timidity package in Nixpkgs is marked as platform `unix` and has special code for darwin so I think it should be ok as is.